aptos-move/framework/src/natives/cryptography/groups/arkworks_backend.rs

```diff
@@ -0,0 +1,1896 @@
+                .unwrap();
+            Ok(NativeResult::ok(
+                gas_params.bls12_381.fq12_serialize,
+                smallvec![Value::vector_u8(buf)],
+            ))
+        }
+        _ => {
+            Ok(NativeResult::err(InternalGas::zero(), E_UNKNOWN_GROUP))
+        }
+    }
+}
+
+fn deserialize_element_uncompressed_internal(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(1, ty_args.len());
+    let type_tag = context
+        .type_to_type_tag(ty_args.get(0).unwrap())?
+        .to_string();
+    let bytes = pop_arg!(args, Vec<u8>);
+    match type_tag.as_str() {
+        "0x1::groups::BLS12_381_G1" => {
+            let cost = gas_params.bls12_381.g1_affine_deserialize_uncompressed + gas_params.bls12_381.g1_affine_to_proj;
+            let point = ark_bls12_381::G1Affine::deserialize_uncompressed(bytes.as_slice());
+            match point {
+                Ok(point) => {
+                    let handle = context
+                        .extensions_mut()
+                        .get_mut::<Bls12381Context>()
+                        .add_g1_point(point.into_projective());
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(true), Value::u64(handle as u64)],
+                    ))
+                }
+                _ => {
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(false), Value::u64(0)],
+                    ))
+                }
+            }
+        }
+        "0x1::groups::BLS12_381_G2" => {
+            let cost = gas_params.bls12_381.g2_affine_deserialize_uncompressed + gas_params.bls12_381.g2_affine_to_proj;
+            let point = ark_bls12_381::G2Affine::deserialize_uncompressed(bytes.as_slice());
+            match point {
+                Ok(point) => {
+                    let handle = context
+                        .extensions_mut()
+                        .get_mut::<Bls12381Context>()
+                        .add_g2_point(point.into_projective());
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(true), Value::u64(handle as u64)],
+                    ))
+                }
+                _ => {
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(false), Value::u64(0)],
+                    ))
+                }
+            }
+        }
+        "0x1::groups::BLS12_381_Gt" => {
+            let cost = (gas_params.bls12_381.fq12_deserialize + gas_params.bls12_381.fq12_pow_fr + gas_params.bls12_381.fq12_eq) * NumArgs::one();
+            let point = Fq12::deserialize(bytes.as_slice());
+            match point {
+                Ok(point) => {
+                    let r_buf = hex::decode(
```

Comment:
You could put this into the PointContext somewhere maybe, to avoid deserializing.

Or use a `static GT_ORDER: Lazy<ark_ff::BigInteger256> = Lazy::new(...);`

Reply:
Fixed.